nil-docs/zkllvm/getting-started/preparing-input-files.mdx

```diff
@@ -0,0 +1,191 @@
+[{ "int": 5 }, { "int": 11 }]
+```
+
+The tutorial circuit only has public input and, therefore, no additional `.inp` files are needed.
+
+The following sub-sections elaborate on the structure and contents of input files. They are optional for completing the tutorial.
+
+## Specifying keys and values
+
+Note that input files only support the following keys (types).
+
+- `"string"` (C-like strings)
+- `"int"` (integers)
+- `"field"` (field elements)
+- `"curve"` (curve elements)
+- `"array"` (`std::array`)
+- `"vector"` (an `ext_vector_type` containing integer, field, or curve elements)
+- `"struct"` (struct or class types)
+
+The table below summarizes what values are accepted by each key.
+
+|                        Key (Type)                         | Values                                                                                               |
+| :-------------------------------------------------------: | :--------------------------------------------------------------------------------------------------- |
+|                        `"string"`                         | Scalar constants                                                                                     |
+| `"int"`, `"field"` and aggregated types that contain them | Scalar constants, strings (in case a number foes not fit within 64 bits), hexadecimal strings        |
+|                         `"curve"`                         | Arrays of numbers containing exactly two values                                                      |
+|             `"array"`, `"vector"`, `"struct"`             | Suitable aggregate types containing exactly the number of elements specified in the type description |
+
+## Extending keys in input files
+
+Except `"int"` and `"string"`, keys can be extended by specifying additional type details in angle brackets (`"<>"`) as shown in the below examples.
+
+- `"field<pallas_scalar>"`
+- `"vector<int>"`
+
+:::info[Aggregate types]
+
+Aggregate types (`"struct"`, `"array"`, and `"vector"`) are extended with the types of the elements they contain. In case of `"array"` and `"vector"`, only a single type needs to be specified. For `"struct"`, it is possible to specify several types separated by a comma.
+
+:::
+
+:::info[Fields and curves]
+
+The `"field"` and `"curve"` keys can be extended by specifying the kind of their elements.
+
+:::
+
+The `"field"` key supports the following extensions.
+
+- `pallas_base`
+- `bls12381_base`
+- `ed25519_base`
+- `pallas_scalar`
+- `bls12381_scalar`
+- `ed25519_scalar`
+
+The `"curve"` key supports the following extensions.
+
+- `pallas`
+- `bls12381`
+- `ed25519`
+
+If an aggregate type is not extended, its elements must be specified as separate key-value pairs in the input file with keys containing their type descriptions.
+
+Consider the following example.
+
+<Tabs groupId='language'>
+  <TabItem value='cpp' label='C++'>
+
+    ```cpp
+    struct PublicInput {
+      arg_one: int;
+      arg_two: int;
+    };
```

Comment:
That's Rust syntax, not C++ :) 

C++ code is:

```cpp
struct PublicInput {
    int arg_one;
    int arg_two;
};
```

Reply:
thank you, fixed!